pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -408,6 +411,58 @@ func (c *Controller) onServiceEvent(_, curr *v1.Service, event model.Event) erro
 	return nil
 }
 
+func serviceEqual(first, second *v1.Service) bool {
```

Comment:
I am very concerned about the cost

Reply:
+1

Reply:
Is the meaning that we shouldn't use `reflect` or something else?

Reply:
@hzxuzhonghu @ramaraochavali I have add `serviceEqual` to the beginning of `onServiceEvent`.  Please take a look again. Thx

Reply:
I think using reflect Equals is bad. If you want to handle conditions, can it be done here https://github.com/istio/istio/blob/161e45ef52fe6ab65f1bec5aad42476163a776f4/pilot/pkg/model/service.go#L1115

Reply:
@ramaraochavali  `Equals` is compare `model.Service`, that have't `conditions`.  In addition, it seems to only take effect after the code below?

Reply:
Annotations and labels are already part of the service comparison I shared above. If you want only conditions - I would say just compare conditions here similar to how the labels and annotations are compared there. But not sure how frequent the condition updates would be

Reply:
> If you want only conditions - I would say just compare conditions here similar to how the labels and annotations are compared there.

I'm not sure if I understand correctly, but as far as I know, there is no Conditions information in model.Service. Therefore, we must compare the Conditions field in the Kubernetes service. 

> But not sure how frequent the condition updates would be

The changes of `Node` configuration will affect the status of the `SVC` when the `SVC` is LB type , so some ServiceController implementations maybe write some information to the conditions of the `SVC`.

@howardjohn